pkg/deploy/api/register.go

```diff
@@ -1,47 +1,83 @@
 package api
 
 import (
+	kapi "k8s.io/kubernetes/pkg/api"
 	"k8s.io/kubernetes/pkg/api/unversioned"
+	"k8s.io/kubernetes/pkg/apis/extensions"
 	"k8s.io/kubernetes/pkg/runtime"
+	"k8s.io/kubernetes/pkg/watch/versioned"
 )
 
-const GroupName = ""
-const FutureGroupName = "deploy.openshift.io"
+const (
+	LegacyGroupName = ""
+	GroupName       = "apps.openshift.io"
```

Comment:
we want to use the same name for the group as kube... it seems like kube also does not stricly follow the plural form for api groups (https://github.com/kubernetes/kubernetes/tree/master/pkg/apis)

I don't see a reason why we should keep trying to do the same. But will leave the decision to @smarterclayton :)